internal/services/arckubernetes/arc_kubernetes_provisioned_cluster_resource.go

```diff
@@ -0,0 +1,400 @@
+	return arckubernetes.ValidateConnectedClusterID
+}
+
+func (r ArcKubernetesProvisionedClusterResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile("^[-_a-zA-Z0-9]{1,260}$"),
+				"The name of Arc Kubernetes Provisioned Cluster can only include alphanumeric characters, underscores, hyphens, has a maximum length of 260 characters, and must be unique.",
+			),
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"identity": commonschema.SystemAssignedIdentityRequiredForceNew(),
+
+		"azure_active_directory": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"admin_group_object_ids": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type:         pluginsdk.TypeString,
+							ValidateFunc: validation.IsUUID,
+						},
+					},
+
+					"azure_rbac_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  false,
+					},
+
+					"tenant_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.IsUUID,
+					},
+				},
+			},
+		},
+
+		"arc_agent_desired_version": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"arc_agent_auto_upgrade_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r ArcKubernetesProvisionedClusterResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"agent_version": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"distribution": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"infrastructure": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"kubernetes_version": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"offering": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"total_core_count": {
+			Type:     pluginsdk.TypeInt,
+			Computed: true,
+		},
+
+		"total_node_count": {
+			Type:     pluginsdk.TypeInt,
+			Computed: true,
+		},
+	}
+}
+
+func (r ArcKubernetesProvisionedClusterResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ArcKubernetes.ArcKubernetesClient
+
+			var model ArcKubernetesProvisionedClusterModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
```

Comment:
Please be mindful of the ordering of calls here
```suggestion
			client := metadata.Client.ArcKubernetes.ArcKubernetesClient
			subscriptionId := metadata.Client.Account.SubscriptionId
			
			var model ArcKubernetesProvisionedClusterModel
			if err := metadata.Decode(&model); err != nil {
				return fmt.Errorf("decoding: %+v", err)
			}
```

Reply:
fixed